synda/pipeline/generation/llm.py

```diff
@@ -16,44 +17,108 @@ def __init__(self, session: Session, run: Run, step_model: Step):
         self.progress = ProgressManager("GENERATION")
         self.provider = Provider.get(self.config.parameters.provider)
         self.model = self.config.parameters.model
+        self.batch = self.config.parameters.batch
+        self.batch_size = self.config.parameters.batch_size
 
     def execute(self, pending_nodes: list[Node], processed_nodes: list[Node]):
         template = self.config.parameters.template
         occurrences = self.config.parameters.occurrences
         instruction_sets = self.config.parameters.instruction_sets
+        instruction_mode = self.config.parameters.instruction_mode
         pending_nodes = self._build_node_occurrences(pending_nodes, occurrences)
-        prompts = PromptBuilder.build(
-            self.session, template, pending_nodes, instruction_sets=instruction_sets
-        )
+
         result = processed_nodes
-        with self.progress.task(
-            "Generating...",
-            len(pending_nodes) + len(processed_nodes),
-            completed=len(processed_nodes),
-        ) as advance:
-            for node, prompt in zip(pending_nodes, prompts):
-                llm_answer = LLMProvider.call(
-                    self.provider.name,
-                    self.model,
-                    self.provider.api_key,
-                    prompt,
-                    url=self.provider.api_url,
-                    temperature=self.config.parameters.temperature,
+        if not self.batch or self.batch_size is None:
```

Comment:
It’s a bit hard to read. I recommend removing the `batch` parameters as they are redundant with `batch_size` and reversing the if/else condition to avoid double negation.